### DISCUSSION TITLE
群晖上面部署后无法正常工作

### DISCUSSION THREAD

**Discussion by Satiakura** [chaitin/SafeLine#818] [Q&A] [answered]
### 问题描述

如题。初步排查了一圈，定位问题出在数据库模块无法正常工作

附相关日志

[mario.log](https://github.com/chaitin/SafeLine/files/13509058/mario.log)

[docker_safeline-mario.log](https://github.com/chaitin/SafeLine/files/13509059/docker_safeline-mario.log)

[postgresql-2023-11-30_042950.json](https://github.com/chaitin/SafeLine/files/13509060/postgresql-2023-11-30_042950.json)

[webserver.log](https://github.com/chaitin/SafeLine/files/13509162/webserver.log)

[nginx.log](https://github.com/chaitin/SafeLine/files/13509164/nginx.log)

### 版本号

3.13.2

### 复现方法

1、使用的在线安装脚本部署，首先遇到的问题是部分必需目录没有自动创建。
手动创建各个目录后 docker-compose up -d 可以正常启动所有容器

2、进入后台页面后各个请求均报502错误，页面内容无法正常显示，功能无法正常运作。如下图所示

![微信图片_20231130140533](https://github.com/chaitin/SafeLine/assets/29559721/45f43c4a-a9f9-4d86-87bc-6a6b0d2bde85)

3、docker ps 查看 safeline-mario 每隔几分钟会重启一次

4、查看 mario.log 发现连接数据库时始终报 timeout

5、进入 safeline-tengine 容器中ping了一下 postgresql 的ip可以连通

5、手动开启 postgresql 日志记录功能截取日志，保存为 postgresql-2023-11-30_042950.json

6、docker logs safeline-mario 截取日志保存为 docker_safeline-mario.log

### 期望的结果

能正常工作

**Comment by yrluke** [comment]
目前可能群晖系统确实不是我们预期的环境，可能还是得单例解决，我突然有一个想法，是不是一些iptables在群晖上出现了走样呢

**Comment by Satiakura** [comment]
> 目前可能群晖系统确实不是我们预期的环境，可能还是得单例解决，我突然有一个想法，是不是一些iptables在群晖上出现了走样呢

如果贵方愿意提供技术支持的话我可以提供力所能及的帮助来解决此问题，毕竟用NAS就是为了脱离云服务器的依赖；况且NAS上面一般都存放了至关重要的个人信息，对安全性也有要求。
私人访问的话流量不会大，性能完全撑得住。
综合以上几点还是有一定实际使用场景和意义的

关于iptables的想法我不知道该怎么样去验证，我附了一些相关的log日志在主题中，可以下载看一下能不能定位到具体原因

**Comment by Lorna0** [comment]
群晖确实不算我们预期的环境，而且目前项目组也没有精力去支持，建议师傅手动根据日志排查一下，或者看看社区有没有其他了解群晖系统的师傅，知道这个问题是如何发生的。

这个转 discussion 开放讨论吧。

**Comment by wintsa123** [comment]
safeline-mario试试重装改host安装？

**Comment by wintsa123** [comment, marked as answer]
问题解决了，关键在于在群晖的时候作为bridge链接时，postgrep配置文件没有允许
pg_hba.conf文件添加
host    all             all             all                     trust

**Reply by Satiakura** [reply]
师傅牛批。因为我群晖用不了就额外添置了软路由，现在waf部署在了软路由上面。还有其他群晖用户有需要的话可以验证一下看
